src/Config/GlobalSection.php

```diff
@@ -46,40 +47,43 @@ public function __construct(EnvironmentReader $environmentReader, Schema $schema
      */
     public function get(string $name)
     {
-        if (!array_key_exists($name, $this->schema->getDefaults(StageConfigInterface::STAGE_GLOBAL))) {
-            throw new \RuntimeException(sprintf(
-                'Config %s was not defined.',
-                $name
-            ));
-        }
-
         try {
-            return $this->mergeConfig()[$name];
-        } catch (\Exception $exception) {
-            throw new ConfigException(
-                $exception->getMessage(),
-                $exception->getCode(),
-                $exception
-            );
+            $defaults = $this->schema->getDefaults(StageConfigInterface::STAGE_GLOBAL);
+            if (!array_key_exists($name, $defaults)) {
+                throw new ConfigException(
+                    sprintf('Config %s was not defined.', $name),
+                    Error::GLOBAL_CONFIG_NOT_DEFINED
+                );
+            }
+
+            return $this->mergeConfig($defaults)[$name];
+        } catch (FileSystemException $e) {
+            throw new ConfigException($e->getMessage(), Error::GLOBAL_CONFIG_UNABLE_TO_READ_SCHEMA_YAML, $e);
```

Comment:
I would not add such specific error code with file extension. Message already has file name.
Also, I would add this code in Magento\MagentoCloud\Config\Schema::getVariables() or even in Magento\MagentoCloud\Filesystem\Driver\File::fileGetContents()

Reply:
I thought about adding code in Magento\MagentoCloud\Config\Schema::getVariables() but in this case, we won't be able to separate codes by stage `deploy`, `build` etc.
It will be one error code for all stages.